packages/contractkit/src/wrappers/SortedOracles.ts

```diff
@@ -1,21 +1,124 @@
+import { Address, CeloContract, CeloToken, NULL_ADDRESS } from '../base'
 import { SortedOracles } from '../generated/types/SortedOracles'
-import { BaseWrapper, proxyCall, toBigNumber } from './BaseWrapper'
+import {
+  BaseWrapper,
+  CeloTransactionObject,
+  proxyCall,
+  toBigNumber,
+  toTransactionObject,
+} from './BaseWrapper'
+
+export enum MedianRelation {
+  Undefined,
+  Lesser,
+  Greater,
+  Equal,
+}
 
 export interface SortedOraclesConfig {
   reportExpirySeconds: BigNumber
 }
 
+export interface OracleRate {
+  address: Address
+  rate: BigNumber
+  medianRelation: MedianRelation
+}
+
+export interface MedianRate {
+  rate: BigNumber
+}
+
 /**
  * Currency price oracle contract.
  */
 export class SortedOraclesWrapper extends BaseWrapper<SortedOracles> {
+  /**
+   * Gets the number of rates that have been reported for the given token
+   * @param token The CeloToken token for which the Celo Gold exchange rate is being reported.
+   * @return The number of reported oracle rates for `token`.
+   */
+  async numRates(token: CeloToken): Promise<BigNumber> {
+    const tokenAddress = await this.kit.registry.addressFor(token)
+    const response = await this.contract.methods.numRates(tokenAddress).call()
+    return toBigNumber(response)
+  }
+
+  /**
+   * Returns the median rate for the given token
+   * @param token The CeloToken token for which the Celo Gold exchange rate is being reported.
+   * @return The median exchange rate for `token`, expressed as:
+   *   amount of that token / equivalent amount in Celo Gold
+   */
+  async medianRate(token: CeloToken): Promise<MedianRate> {
+    const tokenAddress = await this.kit.registry.addressFor(token)
+    const response = await this.contract.methods.medianRate(tokenAddress).call()
+    return {
+      rate: toBigNumber(response[0]).div(toBigNumber(response[1])),
+    }
+  }
+
+  /**
+   * Checks if the given address is whitelisted as an oracle for the token
+   * @param token The CeloToken token
+   * @param oracle The address that we're checking the oracle status of
+   * @returns boolean describing whether this account is an oracle
+   */
+  async isOracle(token: CeloToken, oracle: Address): Promise<boolean> {
+    const tokenAddress = await this.kit.registry.addressFor(token)
+    return this.contract.methods.isOracle(tokenAddress, oracle).call()
+  }
+
   /**
    * Returns the report expiry parameter.
    * @returns Current report expiry.
    */
   reportExpirySeconds = proxyCall(this.contract.methods.reportExpirySeconds, undefined, toBigNumber)
 
+  /**
+   * Updates an oracle value and the median.
+   * @param token The address of the token for which the Celo Gold exchange rate is being reported.
+   * @param numerator The amount of tokens equal to `denominator` Celo Gold.
+   * @param denominator The amount of Celo Gold that the `numerator` tokens are equal to.
+   */
+  async report(
+    token: CeloToken,
+    numerator: number,
+    denominator: number,
```

Comment:
@mcortesi `numerator` and `denominator` aren't very intuitive, as you pointed out in `getRates`. Do you think it's worth changing these as well? I can see leaving them as-is to match the contract signature. But I could also see changing them to something like `amountToken`, `amountGold`. What do you think?

Reply:
There's an issue about changing fraction to fixidity. #1365 Which is something to be done in a later PR.